src/test/test_job.ts

```
@@ -63,6 +63,80 @@ describe('Job', function() {
     });
   });
 
+  it('should move parent job to waiting-children', async () => {
+    const values = [
+      { idx: 0, bar: 'something' },
+      { idx: 1, baz: 'something' },
+      { idx: 2, qux: 'something' },
+    ];
+
+    const parentQueueName = 'parent-queue';
+
+    let childrenProcessor,
+      parentProcessor,
+      processedChildren = 0;
+    const processingChildren = new Promise<void>((resolve, reject) => [
+      (childrenProcessor = async (job: Job) => {
+        processedChildren++;
+        if (processedChildren === values.length) {
+          resolve();
+        }
+      }),
+    ]);
+
+    const processingParent = new Promise<void>((resolve, reject) => [
+      (parentProcessor = async (job: Job, token: string) => {
+        const movedToWaitingChildren = await job.moveToWaitingChildren(token);
+        const { processed } = await job.getDependencies();
+
+        if (Object.keys(processed).length !== 3) {
```

so basically here, this will be true the first time this job is processed right? and then when it is processed the second time after all children are processed it will be false. Please correct me if my assumption was wrong.

---

yes you are right, the first try it should have children, after it move waiting-children the second try will go to the else block

---

in both this cases there must be some silent errors since the first time the job fails but BullMQ is not able to move to job to failed, and in the other case it completes and then it works. I want to figure out an API where moving the processed job from within a processor is legit without breaking current semantics, since I see other uses cases that are important, for example being able for a job to delay itself if some condition is met during the processing.

---

Sounds also like a good addition to support polling